src/python/TaskWorker/Actions/DataDiscovery.py

```diff
@@ -98,4 +102,12 @@ def formatOutput(self, task, requestname, datasetfiles, locations):
         self.logger.debug('Duplicate lumis found: %d' % (lumi_counter - uniquelumis))
         self.logger.debug('Tot files found: %d' % len(wmfiles))
 
+        datasetLumiList = LumiList(runsAndLumis=datasetLumis)
```

Comment:
Can you put some debug printout before and after this block? I fear it might be computational expensive and we might profit from some detailed information in the future..